helper/resource/testing.go

```diff
@@ -1423,3 +1424,36 @@ func detailedErrorMessage(err error) string {
 		return err.Error()
 	}
 }
+
+// indexesIntoTypeSet is a heuristic to try and identify if a flatmap style
+// string address uses a precalculated TypeSet hash, which are always 10 digits
+// long and are a number
+func indexesIntoTypeSet(key string) bool {
+	for _, part := range strings.Split(key, ".") {
+		if len(part) == 10 {
```

Comment:
`TypeSet` hashes can be any positive integer from `0` to `MaxInt` as `SchemaSetFunc` requires an `int` return type: https://github.com/hashicorp/terraform-plugin-sdk/blob/7952a20c9d3c4c46fb0fa855557b8c556283dd23/helper/schema/schema.go#L310-L312

Here's an example test that include varying length hashes: https://github.com/terraform-providers/terraform-provider-aws/blob/7350b304548ea79dff6a626cfd2041057069509f/aws/resource_aws_lb_listener_rule_test.go#L664-L686

To increase the accuracy of this it might be worth going through each part and instead comparing if its an integer then its integer value being greater than 10 or 100 (or something no one would be reasonably testing a `TypeList` with).

Reply:
Ah thank you for the info there, I agree a value greater than 100 is likely a better catch in this case